src/allocation.rs

```diff
@@ -93,6 +93,31 @@ pub fn zeroed_box<T: Zeroable>() -> Box<T> {
   try_zeroed_box().unwrap()
 }
 
+/// Allocates a `Vec<T>` of length and capacity exactly equal to `length` and all elements zeroed.
+/// 
+/// ## Failure
+///
+/// This fails if the allocation fails, or if a layout cannot be calculated for the allocation.
+pub fn try_zeroed_vec<T: Zeroable>(length: usize) -> Result<Vec<T>, ()> {
```

Comment:
I think this should be implemented using `try_zeroed_boxed_slice(length).into_vec()`. It would substantially reduce the unsafe code at play here.